src/Components/Server/src/Circuits/CircuitRegistry.cs

```diff
@@ -309,9 +309,20 @@ private Task PauseAndDisposeCircuitEntry(DisconnectedCircuitEntry entry)
 
     private async Task PauseAndDisposeCircuitHost(CircuitHost circuitHost, bool saveStateToClient)
     {
-        await _circuitPersistenceManager.PauseCircuitAsync(circuitHost, saveStateToClient);
-        circuitHost.UnhandledException -= CircuitHost_UnhandledException;
-        await circuitHost.DisposeAsync();
+        try
+        {
+            await _circuitPersistenceManager.PauseCircuitAsync(circuitHost, saveStateToClient);
+        }
+        catch (ObjectDisposedException ex)
+        {
+            // Expected when service provider is disposed during circuit cleanup e.g. by forced GC
```

Comment:
I don't understand this bit super well.

Reply:
The test that was failing is forcing GC.
https://github.com/dotnet/aspnetcore/blob/9f85d05eb639afd92f3a7b4ce16ccc5042202852/src/Components/test/testassets/Components.TestServer/RazorComponents/Pages/Redirections/CircularRedirection.razor#L38 
It looks like after that operation (that cleaned DI services), we are trying to do:
https://github.com/dotnet/aspnetcore/blob/ea282bf14ecf044e62492c375a8a08e86d00de87/src/Components/Server/src/Circuits/CircuitPersistenceManager.cs#L25

Which is throwing. I'm not sure how else we can get to such situations, if we don't force GC - which is not a common pattern. However, I don't see any harm letting the exception be observed.

Reply:
I think we want to track the root cause of the objectdisposedexception, as it might be pointing out a real issue. If you are able to reproduce it, can you capture the callstack for the object disposed exception?

Reply:
I cannot reproduce it. I really tried, 1k runs locally and no failure

Reply:
> If you are able to reproduce it, can you capture the callstack for the object disposed exception

We could add more logging of inner exception in the test and count on it to fail again on CI. However, we don't know if it will be on a PR that someone will report the failure. Or will they just hit rerun and be happy that it passed the next time.